controller/engine_controller.go

```diff
@@ -1440,18 +1445,15 @@ func (m *EngineMonitor) restoreBackup(engine *longhorn.Engine, rsMap map[string]
 	}
 
 	mlog.Info("Restoring backup")
+	lastRestoredBackup := ""
+	restoreErrorHandler := handleRestoreError
```

Comment:
What's the purpose of the variable? `handleRestoreError` is only called once?

Reply:
We will call `engineClientProxy.BackupRestore` with a different value of the parameter `lastRestoredBackup` and a different error handler `handleRestoreErrorForCompatibleEngine` for the `cliAPIVersion` (< engineapi.CLIVersionFour) in the current implementation.
This modification is to simplify the codes.